### DISCUSSION TITLE
Convert SQLite to MySQL

### DISCUSSION THREAD

**Discussion by CroaBeast** [games647/FastLogin#1111] [General]
Is there a way to convert a SQLite database to a MySQL one? I'm trying to move my server to a network but the FastLogin.db from my survival server does not work in the proxy FastLogin folder.

How can I properly set up FastLogin along with AuthMe in a network after those plugins were used in a standalone server?

**Comment by CroaBeast** [comment]
Also, how can I auto login players between servers to avoid login again everytime they join another server.

**Reply by TuxCoding** [reply]
The force login data is normally forwarded to every server join. However, this only happens if FastLogin logs you in. The server log shows the behavior of FastLogin. If you could provide the server log, we could explain it.

**Reply by CroaBeast** [reply]
@games647 there you go
[2023-11-13-3.log](https://github.com/games647/FastLogin/files/13343252/2023-11-13-3.log)

**Reply by TuxCoding** [reply]
Could you please highlight the player, time and log messages when it happens including from your proxy. Here it is hard to differentiate premium players.

**Comment by TuxCoding** [comment]
There is no official migration method, but you can do it manually using programs like SQLiteman to export your existing data and import it into MySQL using PHPMyAdmin or CLI.

**Reply by CroaBeast** [reply]
is there a tutorial on how to use them? those programs seem complicated

**Reply by TuxCoding** [reply]
1. Let FastLogin generate the table scheme on MariaDB/MySQL
3. Export your existing data to SQL file
   Here only the data is interesting, so select data only and the premium table.
![fastlogin-export](https://github.com/games647/FastLogin/assets/1957196/994f90e9-999e-488f-a4ab-0c575516ff75)
4. Translate to the MySQL language
   Unfortunately, there are some language difference. Therefore replace `BEGIN TRANSACTION` -> `START TRANSACTION` and remove double quotes in the table name if existing.
5. Import the file to MySQL/MariaDB through PHPMyAdmin by selecting first the correct database on the left side and then click on the import tab, where you can upload the file.
![fastlogin-import](https://github.com/games647/FastLogin/assets/1957196/0ba3e88e-9084-4d7f-8b33-ea6946e2d006)
